File: apps/net/Ch3_Lang/L02_iteration.cs
using System;

namespace Ch3_Lang
{
    class Iteration
    {
        public static void Run()
        {
            // WHILE
            string name = "EMPTY";
            while (name?.Length > 0)
            {
                Console.Write("What is your name? ");
                name = Console.ReadLine();
                Console.WriteLine($"Nice to meet you {name}!");
            }

            // foreach
            int[] nums = { 1, 5, 8, 10, 7, 2 };
            foreach (int n in nums)
            {
                Console.WriteLine($"The next number is {n}.");
            }

            // for
            for (int idx = 0; idx < nums.Length; idx++)
            {
                int n = nums[idx];
                Console.WriteLine($"The {idx + 1}th number is {n}.");
            }
        }
    }
}

File: apps/py/ch03_lang/L02_iteration.py
def main():
    print('Python iteration demo')

    # while True:
    #     name = input("What is your name? ")
    #     if not name:
    #         break
    #
    #     print(f"Nice to meet you {name}.")

    nums = [1, 5, 8, 10, 7, 2]  # <-- List<object>
    for n in nums:
        print(f'The next number is {n}.')
    print()

    for idx, n in enumerate(nums, start=1):
        print(f'The {idx}th number is {n}.')

    # NO for (i=0; i < len(nums); i++)
    print()
    for _ in range(1, 6):
        print('This time!')


if __name__ == '__main__':
    main()
